python/tvm/topi/arm_cpu/conv2d_alter_op.py

```diff
@@ -365,3 +439,164 @@ def _alter_conv2d_layout(attrs, inputs, tinfos, out_type):
+    # input channel to be a multiple of 8 and output channels to be a multiple of 8. For input
+    # channels, we pad both the inputs and weights input channels. For output channels, we pad the
+    # weight and stride_slice the output.
+    if is_int8_hw_support(data_dtype, kernel_dtype):
+        # Flags to remember if the expr is modified
+        ic_modified = False
+        oc_modified = False
+
+        # Find the value of input and output channel.
+        in_channel = -1
+        out_channel = -1
+        if attrs["data_layout"] == "NHWC" and attrs["kernel_layout"] == "HWIO":
+            in_channel = data_tensor.shape[3].value
+            out_channel = kernel_tensor.shape[3].value
+        elif attrs["data_layout"] == "NCHW" and attrs["kernel_layout"] == "OIHW":
+            in_channel = data_tensor.shape[1].value
+            out_channel = kernel_tensor.shape[0].value
+        else:
+            return None
+
+        if in_channel % 8 != 0:
+            new_in_channel = ((in_channel + 8) // 8) * 8
+            diff = new_in_channel - in_channel
+            if attrs["data_layout"] == "NHWC" and attrs["kernel_layout"] == "HWIO":
+                data = relay.nn.pad(data, pad_width=((0, 0), (0, 0), (0, 0), (0, diff)))
+                kernel = relay.nn.pad(kernel, pad_width=((0, 0), (0, 0), (0, diff), (0, 0)))
+                ic_modified = True
+            elif attrs["data_layout"] == "NCHW" and attrs["kernel_layout"] == "OIHW":
+                pad_width = ((0, 0), (0, diff), (0, 0), (0, 0))
+                data = relay.nn.pad(data, pad_width=pad_width)
+                kernel = relay.nn.pad(kernel, pad_width=pad_width)
+                ic_modified = True
+            else:
+                return None
+
+        new_out_channel = out_channel
+        if out_channel % 8 != 0:
+            new_out_channel = ((out_channel + 8) // 8) * 8
+            diff = new_out_channel - out_channel
+            if attrs["data_layout"] == "NHWC" and attrs["kernel_layout"] == "HWIO":
+                kernel = relay.nn.pad(kernel, pad_width=((0, 0), (0, 0), (0, 0), (0, diff)))
+                oc_modified = True
+            elif attrs["data_layout"] == "NCHW" and attrs["kernel_layout"] == "OIHW":
+                kernel = relay.nn.pad(kernel, pad_width=((0, diff), (0, 0), (0, 0), (0, 0)))
+                oc_modified = True
+            else:
+                return None
+
+        if oc_modified:
+            new_attrs["channels"] = new_out_channel
+            out = tvm.relay.nn.conv2d(data, kernel, **new_attrs)
+            original_out_shape = [x.value for x in output_tensor.shape]
+            out = relay.strided_slice(out, begin=[0, 0, 0, 0], end=original_out_shape)
+        else:
+            out = relay.nn.conv2d(data, kernel, **new_attrs)
+
+        if is_int8_inputs:
+            out = after_shift(out, adjust_shift)
+
+        return out
+    return None
```

Comment:
Looks like there is a large dup between this change and `x86/conv2d_alter_op.py`... maybe you can clean them up?

Reply:
I moved the common code into `topi/generic/conv2d.py`